src/laboratory/forms.py

```diff
@@ -863,3 +863,101 @@ class Meta:
+                                        label=_("Molecular formula"))
+
+    cas_id_number = forms.CharField(widget=genwidgets.TextInput, required=False,
+                                         label=_("Cas ID Number"))
+
+    security_sheet = forms.FileField(widget=FileChunkedUpload, required=False,
+                                 label=_("Security sheet"))
+
+    is_precursor = forms.BooleanField(widget=genwidgets.YesNoInput,
+                                              required=False,
+                                        label=_("Is Precursor?"))
+
+    precursor_type = forms.ModelChoiceField(widget=genwidgets.Select,
+                                            queryset=Catalog.objects.filter(key='Precursor').using(settings.READONLY_DATABASE),
+                                            blank=True, required=False,
+                                            label=_("Precursor type"))
+
+    h_code = forms.ModelMultipleChoiceField(widget=genwidgets.SelectMultiple,
+                                            queryset=DangerIndication.objects.all().using(settings.READONLY_DATABASE),
+                                            blank=True, required=False,
+                                            label=_("Danger indication  "))
+
+    ue_code = forms.ModelMultipleChoiceField(widget=genwidgets.SelectMultiple,
+                                            queryset=Catalog.objects.filter(key='ue_code').using(settings.READONLY_DATABASE),
+                                            blank=True, required=False,
+                                            label=_("UE Codes"))
+
+    nfpa = forms.ModelMultipleChoiceField(widget=genwidgets.SelectMultiple,
+                                            queryset=Catalog.objects.filter(key='nfpa').using(settings.READONLY_DATABASE),
+                                            blank=True, required=False,
+                                            label=_("NFPA codes"))
+
+    storage_class = forms.ModelMultipleChoiceField(widget=genwidgets.SelectMultiple,
+                                            queryset=Catalog.objects.filter(key='storage_class').using(settings.READONLY_DATABASE),
+                                            blank=True, required=False,
+                                            label=_("Storage class"))
+
+    seveso_list = forms.BooleanField(widget=genwidgets.YesNoInput,
+                                              required=False,
+                                        label=_("Is Seveso?"))
+
+    img_representation = forms.FileField(widget=FileChunkedUpload, required=False,
+                                 label=_("Susstance representation"))
+
+
+    def __init__(self, *args, **kwargs):
+        modal_id = kwargs.pop('modal_id')
+        laboratory_pk = kwargs.pop('laboratory_pk')
+        super(ReactiveForm, self).__init__(*args, **kwargs)
+        self.fields["model"].required = True
+        laboratory_id = "#id_%s-laboratory" % self.prefix
```

Comment:
You are creating a laboratory_id and organization_id variables but you are not using these variables in this form.